gradio/processing_utils.py

```diff
@@ -462,6 +457,55 @@ def _move_to_cache(d: dict):
     return client_utils.traverse(data, _move_to_cache, client_utils.is_file_obj)
 
 
+def _check_allowed(path: str | Path, check_in_upload_folder: bool):
+    blocks = LocalContext.blocks.get()
+    if blocks is None or not blocks.is_running:
+        return
+
+    import tempfile
```

Comment:
suggest putting this import at top of file

Reply:
Done